src/core/config/indices/option.h

```diff
@@ -9,7 +9,8 @@ namespace config {
 
 // This class is meant for creating options that are collections of indices.
 struct IndicesOption {
-    IndicesOption(std::string_view name, std::string_view description);
+    IndicesOption(std::string_view name, std::string_view description,
+                  typename Option<config::IndicesType>::DefaultFunc calculate_default = nullptr);
```

Comment:
By the way, why are we adding `calculate_default` to constructor and not to `operator()` as it is done for `value_check_func`? Using the same `IndicesOption` object to create options with different `calculate_default` functions seems like a useful thing to do

Reply:
We can add `calculate_default` to `operator()`, I will implement this way in separate commit. By the way, in this way we don't need two constructors

Reply:
advantages: we need only one constructor and don't need "strange cast"